libcyphal_demo/src/main.cpp

```diff
@@ -5,40 +5,196 @@
+        }
+        upd_transport = cetl::get<libcyphal::UniquePtr<libcyphal::transport::udp::IUdpTransport>>(  //
+            std::move(maybe_udp_transport));
+
+        upd_transport->setLocalNodeId(7);
+        upd_transport->setTransientErrorHandler(platform::CommonHelpers::Udp::transientErrorReporter);
+    }
+
+    // 2. Create the presentation layer object.
+    //
+    libcyphal::presentation::Presentation presentation{memory, executor, *upd_transport};
+
+    // 3. Create the node object with name.
+    //
+    auto maybe_node = libcyphal::application::Node::make(presentation);
+    if (const auto* failure = cetl::get_if<libcyphal::application::Node::MakeFailure>(&maybe_node))
+    {
+        std::cerr << "Failed to create node (iface='" << static_cast<cetl::string_view>(iface_params.udp_iface.value())
+                  << "').\n";
+        return 10;
+    }
+    auto node = cetl::get<libcyphal::application::Node>(std::move(maybe_node));
+
+    // 4. Populate the node info.
+    //
+    // The hardware version is not populated in this demo because it runs on no specific hardware.
+    // An embedded node would usually determine the version by querying the hardware.
+    auto& get_info                    = node.getInfoProvider().response();
+    get_info.software_version.major   = VERSION_MAJOR;
+    get_info.software_version.minor   = VERSION_MINOR;
+    get_info.software_vcs_revision_id = VCS_REVISION_ID;
+    //
+    const cetl::string_view node_name{NODE_NAME};
+    get_info.name.resize(node_name.size());
+    (void) std::memmove(get_info.name.data(), node_name.data(), get_info.name.size());
 
-    auto cb = executor.registerCallback([](auto&) {
-        //
-        std::cout << "Callback fired." << "\n";
-    });
-    cb.schedule(Callback::Schedule::Repeat{executor.now(), libcyphal::Duration{1s}});
+    // 5. Bring up registry provider, and expose several registers.
+    //
+    if (const auto failure = node.makeRegistryProvider(application.registry()))
+    {
+        std::cerr << "Failed to create registry provider.\n";
+        return 11;
+    }
+    // Expose `get_info.name` as mutable node description.
+    auto reg_node_desc = application.registry().route(  //
+        "uavcan.node.description",
+        [&memory, &get_info] {
+            //
+            return makeStringValue(memory, libcyphal::application::registry::makeStringView(get_info.name));
+        },
+        [&get_info](const auto& value) {
+            //
+            if (const auto* const str = value.get_string_if())
+            {
+                get_info.name = str->value;
+            }
+            return cetl::nullopt;
+        });
+
+    // 5. Bring up the command execution provider.
+    //
+    auto maybe_exec_cmd_provider = AppExecCmdProvider::make(presentation);
+    if (const auto* failure = cetl::get_if<libcyphal::application::Node::MakeFailure>(&maybe_node))
+    {
+        std::cerr << "Failed to create exec cmd provider.\n";
+        return 12;
```

Comment:
Should we have a Status enum for the return values so that we know what 12 maps to?

Reply:
I though that for demo it's not important, but ok I can define such enum.